azurerm/internal/services/eventhub/eventhub_namespace_resource.go

```diff
@@ -69,8 +69,9 @@ func resourceEventHubNamespace() *pluginsdk.Resource {
 				Required:         true,
 				DiffSuppressFunc: suppress.CaseDifference,
 				ValidateFunc: validation.StringInSlice([]string{
-					string(eventhub.Basic),
-					string(eventhub.Standard),
+					string(namespaces.SkuNameBasic),
+					string(namespaces.SkuNameStandard),
+					string(namespaces.SkuNamePremium),
 				}, true),
```

Comment:
because this is case insensitive here, the checks in the customise diff will need to be too

Reply:
good catch!